Mathlib/Probability/Distributions/Gamma.lean

```diff
@@ -0,0 +1,262 @@
+  _ = 1 := by
+    rw [mul_assoc, div_mul_eq_mul_div, ← mul_assoc]
+    have : r^a * (1/r)^a = 1 := by
+      have : 0 ≤ 1/r := by positivity
+      rw [← mul_rpow hr.le this, mul_div, mul_comm, mul_div_assoc, div_self hr.ne']
+      simp
+    rw [this]
+    have : Gamma a / Gamma a = 1 := div_self (Gamma_pos_of_pos ha).ne'
+    simp [this]
+
+open Measure
+
+/-- The Pdf of the gamma Distribution integrates to 1-/
+@[simp]
+lemma lintegral_gammaPdf_eq_one (a r : ℝ) (ha : 0 < a) (hr : 0 < r) : ∫⁻ x, gammaPdf a r x = 1 := by
+  rw [lintegral_eq_lintegral_Ici_add_Iio (gammaPdf a r) 0, ← ENNReal.toReal_eq_one_iff]
+  have leftSide : ∫⁻ x in Iio 0, gammaPdf a r x = 0 := by
+    rw [set_lintegral_congr_fun measurableSet_Iio
+      (ae_of_all _ (fun x (hx : x < 0) ↦ gammaPdf_of_neg hx)), lintegral_zero]
+  have rightSide :
+      ∫⁻ x in Ici 0, gammaPdf a r x =
+      ∫⁻ x in Ici 0, ENNReal.ofReal (r^a/(Real.Gamma a) * x^(a-1) * rexp (-(r * x))) :=
+    set_lintegral_congr_fun isClosed_Ici.measurableSet
+      (ae_of_all _ (fun x (hx : 0 ≤ x) ↦ gammaPdf_of_nonneg hx))
+  simp only [leftSide, add_zero]
+  rw [rightSide]
+  rw [← integral_eq_lintegral_of_nonneg_ae]
+  · rw [MeasureTheory.integral_Ici_eq_integral_Ioi, pow_exp_integral_Ioi_insert_1 ha hr]
+  · unfold EventuallyLE
+    simp only [Pi.zero_apply]
+    rw [ae_restrict_iff]
+    simp only [mem_Ici]
+    · apply ae_of_all
+      intro x hx
+      positivity
+    · refine measurableSet_le measurable_const ?_
+      refine Measurable.mul ?_ ?_
+      · measurability
+      · apply (Measurable.neg ?_).exp
+        exact measurable_id.const_mul _
+  · have := (@MeasureTheory.restrict_Ioi_eq_restrict_Ici ℝ _ ℙ _ _ 0)
+    rw [← this]
+    apply ContinuousOn.aestronglyMeasurable _ (by measurability)
+    apply ContinuousOn.mul
+    · apply ContinuousOn.mul continuousOn_const
+      apply ContinuousOn.rpow_const continuousOn_id
+      intro x hx
+      apply Or.inl
+      exact ne_of_gt hx
+    · apply ContinuousOn.exp
+      apply ContinuousOn.neg
+      apply ContinuousOn.mul continuousOn_const continuousOn_id
+
+end GammaPdf
+
+open MeasureTheory
+
+/-- Measure defined by the Gamma Distribution -/
+noncomputable
+def gammaMeasure (a r : ℝ) : Measure ℝ :=
+  volume.withDensity (gammaPdf a r)
+
+instance instIsProbabilityMeasureGamma (a r : ℝ) [Fact (0 < a)] [Fact (0 < r)] :
+    IsProbabilityMeasure (gammaMeasure a r) where
+  measure_univ := by simp [gammaMeasure, lintegral_gammaPdf_eq_one a r Fact.out Fact.out]
+
+section GammaCdf
+
+/-- CDF of the Gamma Distribution -/
+noncomputable
+def gammaCdfReal (a r : ℝ) : StieltjesFunction :=
+  cdf (gammaMeasure a r)
+
+lemma gammaCdfReal_eq_integral (a r x : ℝ) [Fact (0 < a)] [Fact (0 < r)] :
+    gammaCdfReal a r x = ∫ x in Iic x, gammaPdfReal a r x := by
+  rw [gammaCdfReal,cdf_eq_toReal]
+  simp only [gammaMeasure, measurableSet_Iic, withDensity_apply]
+  rw [integral_eq_lintegral_of_nonneg_ae]; exact rfl
```

Comment:
Use of `;` in mathlib code is discouraged. Try `refine (integral_eq_lintegral_of_nonneg_ae ?_ ?_).symm` and it doesn't create the `rfl` goal anyway